### PR TITLE
feat: bonsai 🌳

### PR DESCRIPTION
**Pull request by bigint**
### <samp>🤖 Generated by Copilot at e86731e</samp>

This pull request adds support for data availability publications, comments, and mirrors, which are new features that allow users to store and verify their content on a decentralized network. It introduces new GraphQL queries and mutations, types and hooks, and UI components to handle the creation and broadcasting of these entities. It also removes some unnecessary type assertions and updates the schema URL to use the latest API endpoint. The changes affect the `apps/web` and `packages/lens` directories.

## Related issues

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at e86731e</samp>

*  Add data availability feature to publications, which allows users to store and verify their content on a decentralized network ([link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R1),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517L40-R55),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R225-R238),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517L222-R262),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R291-R295),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R303),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R314-R324),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R345-R397),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R588-R592),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517R600),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517L504-R634),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-175403612af948ab5a9aed1b111d24448b156cc856927d47f556e01221a56517L549-R684),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-fe787eba695bfa1650d36a6b877e352afd47e34adc6a7625f92afe5c8677094fL11-R14),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-fe787eba695bfa1650d36a6b877e352afd47e34adc6a7625f92afe5c8677094fR120-R126),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-fe787eba695bfa1650d36a6b877e352afd47e34adc6a7625f92afe5c8677094fR133-R138),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-fe787eba695bfa1650d36a6b877e352afd47e34adc6a7625f92afe5c8677094fL154-R197),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-0e0b11ed8e8f757115c7cba9baab8d7b185c0f178ce22cde62478e3e42494086R52-R58),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-4d5e131db167ad0d98bcfb0aacc2f2679baf99dd3555d6fd1e7a31717cb144aeR11-R12),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-4d5e131db167ad0d98bcfb0aacc2f2679baf99dd3555d6fd1e7a31717cb144aeR49-R50),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-1ab41ae3a982c2b5184a2d6efd41ba0a5574f112c53249159f84080eda0640a2R9-R10),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-1ab41ae3a982c2b5184a2d6efd41ba0a5574f112c53249159f84080eda0640a2R45-R46),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-ede11b1b2d5a078d958ca58815cddfa284b7e3eba70ec4250b6de825ebeaf808R11-R12),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-4bd4016e1fc38be212a292a4dcdbd95ffa147cf521cc5739f67e0278bb5316f8R1-R11),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-afbc724f6a38a7461056d9514e1afbc3468ade92b76e7bb6526e0069df3c72c7R1-R33),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-7fd52f28733c630f9d7a7b9f893680f4fa5d972720c33f1eb45a6a941f53b37cR1-R11),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-fa78ebbee4fec603ca560b591f3f869c2b3fc4b8c0aa51eac94e34f411b61ea2R1-R11),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-604860aeaa09817d46137d6371c9d3d0deeb8be9d827d2e0a9147ccf0fc2085eR1-R30),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-fd35c1b0a2cc2297bcbcb9cb2dbb8da94388fad0bea4d4fa37bddbe20f7223ceR1-R11))
* Update schema URL in `codegen.yml` to use the staging API endpoint for the social network, which supports the data availability feature ([link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-a02d7522dc575a96a81a9d0c0e14d5e7730a583e2258b6fe7d537ccd0a205834L1-R1))
* Remove `as any` type assertions from `profile` and `publication` props and data, and use the `Profile` and `Publication` types from the `lens` package instead, which improve type safety and readability ([link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-d9d21ca04ede0cf97008d0ebda640d5aca84b1c11c8f284f3d03f93a7faa48b0L37-R41),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-d4f9ff371b1d883105df82925421bcfdc69977d6f08eec12245dc58eeb124786L96-R96),[link](https://github.com/lensterxyz/lenster/pull/2122/files?diff=unified&w=0#diff-ed6eceb61013247be2d221cb5265af7a4d8ceffe42846702068115c3f915e81aL47-R47))

## Emoji

📡📝🔒

### CODE REVIEW

File: apps/web/src/components/Profile/index.tsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ const ViewProfile: NextPage = () => {
       />
       <GridLayout className="pt-6">
         <GridItemFour>
-          <Details profile={profile as any} following={!!following} setFollowing={setFollowing} />
+          <Details profile={profile as Profile} following={!!following} setFollowing={setFollowing} />
```
Comment (harish-sethuraman) [inline apps/web/src/components/Profile/index.tsx:117]: why !! is used here you are converting number to boolean?

Comment (bigint) [reply apps/web/src/components/Profile/index.tsx:117]: It's not number its `boolean | null`

Comment (bigint) [reply apps/web/src/components/Profile/index.tsx:117]: Also true `!!` is a good way, I changed to `Boolean(following)`

**Review by saagu335 — APPROVED** [review]
LGTM :shipit:

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 2]
```diff
@@ -42,19 +42,28 @@ import type {
...
+  useCreateDataAvailabilityCommentTypedDataMutation,
+  useCreateDataAvailabilityCommentViaDispatcherMutation,
+  useCreateDataAvailabilityPostTypedDataMutation,
+  useCreateDataAvailabilityPostViaDispatcherMutation,
   useCreatePostTypedDataMutation,
-  useCreatePostViaDispatcherMutation
+  useCreatePostViaDispatcherMutation,
+  usePublicationLazyQuery
 } from 'lens';
+import { useApolloClient } from 'lens/apollo';
```
Comment (saagu335) [inline apps/web/src/components/Composer/NewPublication.tsx:61]: Kudos 👏

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 3]
```diff
@@ -215,6 +227,20 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
...
+  const [broadcastDataAvailability] = useBroadcastDataAvailabilityMutation({
+    onCompleted: (data) => {
+      onCompleted();
+      if (data?.broadcastDataAvailability.__typename === 'RelayError') {
+        return toast.error(Errors.SomethingWentWrong);
+      }
+
+      if (data?.broadcastDataAvailability.__typename === 'CreateDataAvailabilityPublicationResult') {
+        push(`/posts/${data?.broadcastDataAvailability.id}`);
+      }
```
Comment (saagu335) [inline apps/web/src/components/Composer/NewPublication.tsx:239]: Can we make an optimistic push to the feed?

File: packages/lens/codegen.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-schema: https://api-mumbai.lens.dev
+schema: https://staging-api-social-mumbai.lens.crtlkey.com
```
Comment (saagu335) [inline packages/lens/codegen.yml:1]:
```suggestion
schema: https://api-mumbai.lens.dev
```

Reverting because it's deployed in prod testnet

**Review by saagu335 — CHANGES_REQUESTED** [review]
We need to move everything to hook 🙇

File: apps/web/src/components/Composer/NewPublication.tsx (hunk 3 above)
Comment (bigint) [reply apps/web/src/components/Composer/NewPublication.tsx:239]: We can make it happen in next iteration

**Review by saagu335 — APPROVED** [review]

**Pull request merged**